lib/nrf_inbuilt_key/nrf_inbuilt_key.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <at_cmd.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <at_cmd_parser/at_params.h>
+#include <bsd_limits.h>
+#include <nrf_inbuilt_key.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define NRF_INBUILT_KEY_OP_LS "AT%CMNG=1"
+#define NRF_INBUILT_KEY_OP_RD "AT%CMNG=2"
+#define NRF_INBUILT_KEY_OP_WR "AT%CMNG=0"
+#define NRF_INBUILT_KEY_OP_RM "AT%CMNG=3"
+
+#define NRF_INBUILT_KEY_CHAR_SIZE_MAX_INT32 (10)
+#define NRF_INBUILT_KEY_CR_CHAR_SIZE (1)
+#define NRF_INBUILT_KEY_LF_CHAR_SIZE (1)
+#define NRF_INBUILT_KEY_QUOTES_CHAR_SIZE (1)
+#define NRF_INBUILT_KEY_ZERO_TERMINATOR_SIZE (1)
+#define NRF_INBUILT_KEY_COMMA_SIZE (1)
+#define NRF_INBUILT_KEY_INVALID_HANDLE (-1)
+
+static u8_t scratch_buf[4096];
+static char at_cmee_strings[2][10] = {"AT+CMEE=0", "AT+CMEE=1"};
+
+static int cmee_active(void)
+{
+	char response[sizeof("+CMEE: X\r\n") + 1];
+
+	memset(response, 0, sizeof(response));
+
+	enum at_cmd_state state;
+
+	int retval =
+		at_cmd_write("AT+CMEE?", response, sizeof(response), &state);
+	if (retval == 0) {
+		char *deactivated = strchr(&response[6], '0');
```

Comment:
Why `strchr` here? Are you trying to compare `response[6]` with `0`?

Reply:
He's not sure exactly where it is, so he wants to play it safe by searching for it instead of checking a single char.